core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -788,11 +779,15 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
     // FIXME_SYCL so far accessors used instead of these pointers
     // Functor's reduce memory, team scan memory, and team shared memory depend
     // upon team size.
-    const auto& space    = *m_policy.space().impl_internal_space_instance();
-    const sycl::queue& q = *space.m_queue;
-    m_scratch_ptr[0]     = nullptr;
-    m_scratch_ptr[1]     = sycl::malloc_device(
-        sizeof(char) * m_scratch_size[1] * m_league_size, q);
+    const auto& space = *m_policy.space().impl_internal_space_instance();
+    m_scratch_ptr[0]  = nullptr;
+    m_scratch_ptr[1]  = m_team_size <= 0
+                           ? nullptr
+                           : m_policy.space()
+                                 .impl_internal_space_instance()
+                                 ->resize_team_scratch_space(
+                                     static_cast<ptrdiff_t>(m_scratch_size[1]) *
+                                     m_league_size);
```

Comment:
These are `int` right?  Why do you cast?

Reply:
Purely conformity with the `Cuda` implementation. I can change that if you want.